lib/LaTeXML/Package/fancyhdr.sty.ltxml

```diff
@@ -24,46 +24,56 @@ use LaTeXML::Package;
 
 ######################################################
 
-DefMacroI('\lhead', '{}', '');
-DefMacroI('\chead', '{}', '');
-DefMacroI('\rhead', '{}', '');
-
-DefMacroI('\lfoot', '{}', '');
-DefMacroI('\cfoot', '{}', '');
-DefMacroI('\rfoot', '{}', '');
-
 DefMacroI('\fancyhead', '[]{}', '');
 DefMacroI('\fancyfoot', '[]{}', '');
-
-DefMacroI('\fancypagestyle', '{}{}', '');
-DefMacroI('\fancyhf',        '[]',   '');
+DefMacroI('\fancyhf',   '[]{}', '');
 
 DefMacroI('\fancyheadoffset', '[]{}', '');
 DefMacroI('\fancyfootoffset', '[]{}', '');
 DefMacroI('\fancyhfoffset',   '[]{}', '');
 
-DefMacroI('\iffloatpage', '{}{}', '');
-DefMacroI('\iftopfloat',  '{}{}', '');
-DefMacroI('\ifbotfloat',  '{}{}', '');
+DefMacroI('\headrulewidth', undef, '0.4pt');
+DefMacroI('\footrulewidth', undef, '0pt');
+DefMacroI('\headruleskip',  undef, '0pt');                     # since 4.0
+DefMacroI('\footruleskip',  undef, '.3\normalbaselineskip');
+DefMacroI('\headrule',      undef, '');
+DefMacroI('\footrule',      undef, '');
+DefRegister('\headwidth' => Dimension(0));                     # maybe need some other value here?
 
-DefConditional('\iffootnote', undef);
+DefMacroI('\fancyheadinit', '{}', '');                         # since 4.0
+DefMacroI('\fancyfootinit', '{}', '');                         # since 4.0
+DefMacroI('\fancyhfinit',   '{}', '');                         # since 4.0
 
-DefMacroI('\headrulewidth',      undef, '0.4pt');
-DefMacroI('\footrulewidth',      undef, '0pt');
-DefMacroI('\plainheadrulewidth', undef, '0pt');
-DefMacroI('\plainfootrulewidth', undef, '0pt');
-DefMacroI('\footruleskip',       undef, '.3\normalbaselineskip');
+# not implemented yet: \fancycenter[][]{}{}{}, since 4.0
+
+# always false as LaTeXML does not paginate
+DefMacroI('\iffloatpage', '{}{}', '#2');
+DefMacroI('\iftopfloat',  '{}{}', '#2');
+DefMacroI('\ifbotfloat',  '{}{}', '#2');
+DefMacroI('\iffootnote',  '{}{}', '#2');    # since 3.8
+# DefConditional('\iffootnote', undef);     # pre-3.8 implementation
```

Comment:
This could break old files! How do you deal with this kind of breaking changes?

Reply:
Hmm, fancyhdr (currently) defines them as macros in terms of `\ifx`. It seems dubious style to define `\iffootnote{}{}` as a regular macro, rather than real conditional `\iffootnote#1\else#2\fi`, but it seems like it would be a major change to redefine it that way from one version to the next. I don't see any hint of that in the documentation. Are you sure it ever was a conditional (ie. `\newif`)?

I suppose there could be corner cases where your definition above would work differently than an expansion in terms of `\ifx`. If that was a concern, you could copy fancyhdr's defns, and add your own things like
```
DefConditional('\if@fcolmode');
DefConditional('\iff@nch@footnote');
Let('\topfloat','\empty');
Let('\botfloat','\empty');
```
(those might be good to define, anyway)

Reply:
fancyhdr really did a breaking change in 2016, `\iffootnote` was a conditional and was changed into `\iffootnote{}{}`, see https://tug.org/svn/texlive/trunk/Master/texmf-dist/tex/latex/fancyhdr/fancyhdr.sty?r1=41944&r2=42004.

I can't see a way out, other than reading `fancyhdr.sty` itself?

Reply:
Wow. Reading `fancyhdr.sty` would probably just get you the new version anyway! :>  Luckily only `iffootnote` got this special treatment. You could just go with the current behaviour and let the few old uses of `\iffootnote` break.  If you wanted to get fancy, you might gamble that there's always an opening '{', so that something like
```
DefMacro('\iffootnote`,sub {
  return ($_[0]->ifNext(T_BEGIN) ? T_CS('\lx@iffootnote@macro') : T_CS('\lx@iffootnote@cond`)); });
```
with the 2 latter being the new & old defns, respectively.
(You might look at `OmniBus.cls.ltxml`, which is a collection of dirty tricks, trying to be a class file for all times & places)

Reply:
If you are ok with the breakage, maybe it's better to go with the current behaviour. Hopefully the old uses are all within headers and footers. I am thinking, the hack can get tricky to debug when the heuristics is wrong.

Ideally, I would like to extract the version number. Another hack could be to redefine `\ProvidesPackage` so that it saves the version number and calls `\endinput`, then reads `fancyhdr.sty` (or some alternative form of grepping). Too complicated?

Reply:
I think I'm fine with a little potential breakage. As you note: if it's used at all, it will likely be wthin one of the header macros which will never be expanded anyway.